http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -68,6 +78,20 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
 
       _ <- checkIfUserAllowed(req.appType, userInfo.userEmail)
 
+      projectLabels <- googleResourceService.getLabels(googleProject)
+      appAllowList =
+        if (projectLabels.getOrElse("security-group", "").canEqual("high"))
```

Comment:
can we make the string `security-group` and `high` a constant at the top of this class..

Also can we move the code into `checkIfUserAllowed` instead.

The check here isn't quite right. So if there's no project label for `security-group`, we want to keep using the previous check; only when there is `security-group` label, we'll enforce this check.

Another thing is before we check both, we should check `enableCustomAppCheck` flag first